delegation-toolkit/concepts/erc7715.md

```diff
@@ -0,0 +1,62 @@
+---
+description: Learn about MetaMask ERC-7715 permissions.
+keywords: [ERC-7715, 7715, permissions, wallet, smart account]
+---
+
+# ERC-7715 permissions
+
+The Delegation Toolkit supports [ERC-7715](https://eips.ethereum.org/EIPS/eip-7715), which lets you request fine-grained permissions from a MetaMask user to execute transactions on their behalf.
+For example, a user can grant your dapp permission to spend 10 USDC per day to buy ETH over the course of a month.
+Once the permission is granted, your dapp can use the allocated 10 USDC each day to purchase ETH directly from the MetaMask user's account.
+
+ERC-7715 eliminates the need for users to approve every transaction, which is useful for highly interactive dapps.
+It also enables dapps to execute transactions for users without an active wallet connection.
+
+:::note
+This feature requires [MetaMask Flask 12.14.2](/snaps/get-started/install-flask) or later.
+:::
+
+## ERC-7715 technical overview
+
+[ERC-7715](https://eips.ethereum.org/EIPS/eip-7715) defines a JSON-RPC method `wallet_grantPermissions`. 
+Dapps can use this method to request a wallet to grant the dapp permission to execute transactions on a user's behalf.
+`wallet_grantPermissions` requires a `signer` parameter, which identifies the entity requesting or managing the permission.
+Common signer implementations include wallet signers, single key and multisig signers, and account signers.
+
+The Delegation Toolkit's implementation uses an account signer.
+A session account is created and used for the single purpose of requesting and redeeming ERC-7715 permissions, and does not contain tokens.
+The session account can be granted with permissions and redeem them as specified in [ERC-7710](https://eips.ethereum.org/EIPS/eip-7710).
+The session account can be a smart account or an externally owned account (EOA).
+
+The MetaMask user that the session account requests permissions from must be upgraded to a [MetaMask smart account](smart-accounts.md).
+
+## ERC-7715 vs. delegations
+
+ERC-7715 expands on regular [delegations](delegation/index.md) by enabling permission sharing *via the MetaMask browser extension*.
+
+With regular delegations, the dapp constructs a delegation and requests the user to sign it.
+These delegations are not human-readable, so it is the dapp's responsibility to provide context for the user.
+Regular delegations cannot be signed through the MetaMask extension, because if a dapp requests a delegation without constraints, the whole wallet can be exposed to the dapp.
```

Comment:
"Regular delegations cannot be signed for MetaMask extension account", because

Reply:
Or maybe something that says that it's only disabled for internal accounts. So if we use the MM account as an signer for the Smart Account, we can still sign the delegations.